src/Service.Tests/Unittests/ConfigFileWatcherUnitTests.cs

```diff
@@ -0,0 +1,150 @@
+      ""path"": """ + initialGQLPath + @""",
+      ""allow-introspection"": " + initialGQLIntrospection.ToString().ToLower() + @"
+    },
+    ""host"": {
+      ""cors"": {
+        ""origins"": [
+          ""http://localhost:5000""
+        ],
+        ""allow-credentials"": false
+      },
+      ""authentication"": {
+        ""provider"": ""StaticWebApps""
+      },
+      ""mode"": """ + initialMode + @"""
+    }
+  },
+  ""entities"": {}
+}";
+            string updatedConfig = @"
+{
+  ""$schema"": ""https://github.com/Azure/data-api-builder/releases/download/vmajor.minor.patch/dab.draft.schema.json"",
+  ""data-source"": {
+    ""database-type"": ""mssql"",
+    ""connection-string"": ""Server=test;Database=test;User ID=test"",
+    ""options"": {
+      ""set-session-context"": true
+    }
+  },
+  ""runtime"": {
+    ""rest"": {
+      ""enabled"": " + updatedRestEnabled.ToString().ToLower() + @",
+      ""path"": """ + updatedRestPath + @"""
+    },
+    ""graphql"": {
+      ""enabled"": " + updatedGQLEnabled.ToString().ToLower() + @",
+      ""path"": """ + updatedGQLPath + @""",
+      ""allow-introspection"": " + updatedGQLIntrospection.ToString().ToLower() + @"
+    },
+    ""host"": {
+      ""cors"": {
+        ""origins"": [
+          ""http://localhost:5000""
+        ],
+        ""allow-credentials"": false
+      },
+      ""authentication"": {
+        ""provider"": ""StaticWebApps""
+      },
+      ""mode"": """ + updatedMode + @"""
+    }
+  },
+  ""entities"": {}
+}";
+            string configName = "hotreload-config.json";
+            if (File.Exists(configName))
+            {
+                File.Delete(configName);
+            }
+
+            FileSystem fileSystem = new();
+            fileSystem.File.WriteAllText(configName, initialConfig);
+            FileSystemRuntimeConfigLoader configLoader = new(fileSystem, configName, string.Empty);
+            RuntimeConfigProvider configProvider = new(configLoader);
+            // Must GetConfig() to start file watching
+            RuntimeConfig runtimeConfig = configProvider.GetConfig();
+            // assert we have a valid config
+            Assert.IsNotNull(runtimeConfig);
+            Assert.AreEqual(initialRestEnabled, runtimeConfig.Runtime.Rest.Enabled);
+            Assert.AreEqual(initialRestPath, runtimeConfig.Runtime.Rest.Path);
+            Assert.AreEqual(initialGQLEnabled, runtimeConfig.Runtime.GraphQL.Enabled);
+            Assert.AreEqual(initialGQLPath, runtimeConfig.Runtime.GraphQL.Path);
+            Assert.AreEqual(initialGQLIntrospection, runtimeConfig.Runtime.GraphQL.AllowIntrospection);
+            Assert.AreEqual(initialMode, runtimeConfig.Runtime.Host.Mode);
+            // Simulate change to the config file
+            fileSystem.File.WriteAllText(configName, updatedConfig);
+            // Give file watcher enough time to see the change
+            System.Threading.Thread.Sleep(1000);
```

Comment:
I am not sure what you mean here. We have a FileWatcher that will raise an event when the config file changes, but that is within the config file watcher class. Do you mean to add another file watcher inside of this test that then can raise another event for us to wait on?

I am not sure this is worth the effort because we still need to thread.sleep in order to give the hot reload enough time to take place. We could in theory have an event that gets signaled when the hot reload is finished, but that would be an event within our config file watcher code that we signal only for testing.

It seems like for a simple unit test, simply sleeping for 1 second is a more natural approach to me.

Reply:
My intent was to mitigate the need to have an explicit thread.sleep. While it is 1 second and 1 test now, that wouldn't really be scalable if more tests had to do that. 

The current comment you have above the sleep code indicates that the wait is required because it may take time for the file changed event to be fired and received by the file watcher. 

If that's not the case and the wait is because it takes a certain amount of time to complete `hotReload()` related activities like recreating the runtime config object, then the comment should be updated.

My suggestion was to have another event listener in this test that listens for the file changed event to be fired by the filesystem, assuming the thread.sleep was used because of latency between filechanged notification being fired and filewatcher processing that notification. In other words, "wait for the filechanged notification and right when we detect that, begin our asserts" Ultimately, that doesn't address your latest point of the time required to finish `hotreload()`

Reply:
I see what you mean, I'll update the comment so it is more clear that we really need to wait on the event plus the actual hot reloading.